[visionOS] no VP8 or VP9 available in visionOS VM.

jyavenard opened the pull request:
#### 77966f5b8e272f8b790a223fc21c036dcf857b1c
<pre>
[visionOS] no VP8 or VP9 available in visionOS VM.
<a href="https://bugs.webkit.org/show_bug.cgi?id=275204">https://bugs.webkit.org/show_bug.cgi?id=275204</a>
<a href="https://rdar.apple.com/129329733">rdar://129329733</a>

Reviewed by Youenn Fablet.

VP8 and VP9 codecs are currently not available on visionOS VM making testing difficult.
The actual hardware has HW acceleration for VP9 and VP8 is available in VideoToolbox.

We add an internal web preference to always enable software decoders.
On machines with no VP8/VP9 decoder we will fallback on the WebCoreDecompressionSession and
the WebRTC software decoders.

There&apos;s no current testing infrastructure for this configuration.
The underlying bug is tracked under
<a href="https://rdar.apple.com/129329733">rdar://129329733</a>

* LayoutTests/media/vp9.html: rename preference.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::setSWDecoderAlwaysEnabled):
(WebCore::PlatformMediaSessionManager::swDecoderAlwaysEnabled):
(WebCore::PlatformMediaSessionManager::setShouldEnableVP9SWDecoder): Deleted.
(WebCore::PlatformMediaSessionManager::shouldEnableVP9SWDecoder): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::ensureCodecsRegistered):
* Source/WebCore/platform/graphics/cocoa/VP9UtilitiesCocoa.h:
* Source/WebCore/platform/graphics/cocoa/VP9UtilitiesCocoa.mm:
(WebCore::VP9TestingOverrides::setSWDecoderAlwaysEnabled):
(WebCore::isSWDecoderAlwaysEnabled):
(WebCore::isVP9DecoderAvailable):
(WebCore::isVP8DecoderAvailable):
(WebCore::isVP9CodecConfigurationRecordSupported):
(WebCore::computeVPParameters):
* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::updateGPUProcessPreferences):
* Source/WebKit/GPUProcess/GPUProcess.h:
* Source/WebKit/GPUProcess/GPUProcessPreferences.cpp:
(WebKit::GPUProcessPreferences::copyEnabledWebPreferences):
* Source/WebKit/GPUProcess/GPUProcessPreferences.h:
* Source/WebKit/GPUProcess/GPUProcessPreferences.serialization.in:
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/Shared/WebPageCreationParameters.serialization.in:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::creationParameters):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_unifiedTextReplacementController):

Canonical link: <a href="https://commits.webkit.org/279806@main">https://commits.webkit.org/279806@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/350c90d790c0e66ff8baa8ab0a8e73bb83edddd9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54610 "4 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34036 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7183 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57891 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5344 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/56912 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41569 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5347 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44229 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3600 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56705 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32174 "Found 60 new test failures: accessibility/accessibility-node-reparent.html accessibility/ancestor-computation.html accessibility/announcement-notification.html accessibility/aria-braillelabel.html accessibility/aria-brailleroledescription.html accessibility/aria-checked-mixed-value.html accessibility/aria-current-state-changed-notification.html accessibility/aria-current.html accessibility/aria-describedby-on-input.html accessibility/aria-description.html ... (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47301 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25355 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28978 "Found 60 new test failures: imported/w3c/web-platform-tests/FileAPI/Blob-methods-from-detached-frame.html imported/w3c/web-platform-tests/FileAPI/FileReader/Progress_event_bubbles_cancelable.html imported/w3c/web-platform-tests/FileAPI/FileReader/workers.html imported/w3c/web-platform-tests/FileAPI/FileReaderSync.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-array-buffer.any.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-array-buffer.any.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-bytes.any.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-bytes.any.worker.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-constructor-dom.window.html imported/w3c/web-platform-tests/FileAPI/blob/Blob-constructor-endings.html ... (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4632 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3484 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/47975 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/50720 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4848 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59481 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/54117 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29845 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/4987 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51653 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30993 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47387 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51030 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11954 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31973 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/66414 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30781 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12663 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
```diff
@@ -813,14 +816,17 @@ bool PlatformMediaSessionManager::shouldEnableVP8Decoder()
 }
 
-bool PlatformMediaSessionManager::shouldEnableVP9SWDecoder()
+bool PlatformMediaSessionManager::swDecoderAlwaysEnabled()
```
```suggestion
bool PlatformMediaSessionManager::swDecoderAlwaysEnabled() const
```

jyavenard replied:
The method is static

youennf commented on Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
```diff
@@ -59,7 +62,7 @@ bool PlatformMediaSessionManager::s_useSCContentSharingPicker;
 bool PlatformMediaSessionManager::m_vp9DecoderEnabled;
 bool PlatformMediaSessionManager::m_vp8DecoderEnabled;
-bool PlatformMediaSessionManager::m_vp9SWDecoderEnabled;
+bool PlatformMediaSessionManager::m_swDecoderAlwaysEnabled;
```
```suggestion
bool PlatformMediaSessionManager::m_swVPDecodersEnabled;
```

youennf replied:
Ditto for setter/getter

youennf commented on Source/WebKit/GPUProcess/GPUProcess.cpp:
```diff
@@ -360,12 +360,19 @@ void GPUProcess::updateGPUProcessPreferences(GPUProcessPreferences&& preferences
-            m_haveEnabledVP9SWDecoder = true;
-            WebCore::registerWebKitVP9Decoder();
+        if (!m_haveEnabledSWDecoder && m_preferences.swDecoderAlwaysEnabled) {
+            m_haveEnabledSWDecoder = true;
```
```suggestion
            m_haveEnabledSWDecoders = true;
```

jyavenard committed:
[visionOS] no VP8 or VP9 available in visionOS VM.
https://bugs.webkit.org/show_bug.cgi?id=275204
rdar://129329733

Reviewed by Youenn Fablet.

VP8 and VP9 codecs are currently not available on visionOS VM making testing difficult.
The actual hardware has HW acceleration for VP9 and VP8 is available in VideoToolbox.

We add an internal web preference to always enable software decoders.
On machines with no VP8/VP9 decoder we will fallback on the WebCoreDecompressionSession and
the WebRTC software decoders.

There's no current testing infrastructure for this configuration.
The underlying bug is tracked under
rdar://129329733

* LayoutTests/media/vp9.html: rename preference.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::setSWDecoderAlwaysEnabled):
(WebCore::PlatformMediaSessionManager::swDecoderAlwaysEnabled):
(WebCore::PlatformMediaSessionManager::setShouldEnableVP9SWDecoder): Deleted.
(WebCore::PlatformMediaSessionManager::shouldEnableVP9SWDecoder): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::ensureCodecsRegistered):
* Source/WebCore/platform/graphics/cocoa/VP9UtilitiesCocoa.h:
* Source/WebCore/platform/graphics/cocoa/VP9UtilitiesCocoa.mm:
(WebCore::VP9TestingOverrides::setSWDecoderAlwaysEnabled):
(WebCore::isSWDecoderAlwaysEnabled):
(WebCore::isVP9DecoderAvailable):
(WebCore::isVP8DecoderAvailable):
(WebCore::isVP9CodecConfigurationRecordSupported):
(WebCore::computeVPParameters):
* Source/WebKit/GPUProcess/GPUProcess.cpp:
(WebKit::GPUProcess::updateGPUProcessPreferences):
* Source/WebKit/GPUProcess/GPUProcess.h:
* Source/WebKit/GPUProcess/GPUProcessPreferences.cpp:
(WebKit::GPUProcessPreferences::copyEnabledWebPreferences):
* Source/WebKit/GPUProcess/GPUProcessPreferences.h:
* Source/WebKit/GPUProcess/GPUProcessPreferences.serialization.in:
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/Shared/WebPageCreationParameters.serialization.in:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::creationParameters):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_unifiedTextReplacementController):

Canonical link: https://commits.webkit.org/279806@main
